src/groups/circle_group.jl

```diff
@@ -0,0 +1,58 @@
+@doc doc"""
```

Comment:
So this is the same as the real-valued circle with the addition group? I think that could just shortly be mentioned, since we have the addition group anyways.

Reply:
Without overloading `compose` for the real circle group case to also take the modulo, then `AdditionOperation` applied to the real circle will not be the same group.
The "right" way to do this would be to have
```julia
struct CircleGroup{𝔽,O} <: AbstractGroupManifold{O}
    manifold::Circle{𝔽}
end
```
and then have different operations for the two cases. But then none of the `GroupManifold` forwards apply. Forwarding metric functions from `AbstractGroupManifold` is no less difficult than implementing the general forwarding default proposed in #84.

But I also think we should interpret `AdditionOperation` and `MultiplicationOperation` literally as being `+` and `-`, so a new operation `AdditionWithModulo` would be necessary.

Reply:
Yes, you're right, it's actually addition modulo, sorry.